Sources/SignalRClient/Transport/WebSocketTransport.swift

```diff
@@ -3,168 +3,222 @@ import Foundation
+        // When receive websocket close message?
+        nonisolated func urlSession(_ session: URLSession, webSocketTask: URLSessionWebSocketTask, didCloseWith closeCode: URLSessionWebSocketTask.CloseCode, reason: Data?) {
+            logger.log(level: .debug, message: "(WebSockets transport) URLSession didCloseWith: \(closeCode)")
 
-        Task {
-            try await stop(error: nil)
+            Task {
+                await stop(error: nil)
+            }
         }
-    }
 
-    public func urlSession(_ session: URLSession, webSocketTask: URLSessionWebSocketTask, didOpenWithProtocol protocol: String?) {
-        logger.log(level: .debug, message: "(WebSockets transport) urlSession didOpenWithProtocol invoked. WebSocket open")
-    }
+        nonisolated func urlSession(_ session: URLSession, webSocketTask: URLSessionWebSocketTask, didOpenWithProtocol protocol: String?) {
+            logger.log(level: .debug, message: "(WebSockets transport) urlSession didOpenWithProtocol invoked. WebSocket open")
 
-    private func receiveMessage() async {
-        guard let websocket = websocket else {
-            logger.log(level: .error, message: "(WebSockets transport) WebSocket is nil")
-            return 
+            Task {
+                if await openTcs.trySetResult(.success(())) == true {
+                    logger.log(level: .debug, message: "(WebSockets transport) WebSocket connected")
+                }
+            }
         }
-        
-        logger.log(level: .error, message: "(WebSockets transport) Start receiving messages")
-
-        do {
-            while true {
-                let message = try await websocket.receive()
-
-                switch message {
-                    case .string(let text):
-                        logger.log(level: .debug, message: "(WebSockets transport) Received message: \(text)")
-                        await onReceive?(.string(text))
-                    case .data(let data):
-                        await onReceive?(.data(data))
+
+        private func receiveMessage() async {
+            guard let websocket: URLSessionWebSocketTask = websocket else {
+                logger.log(level: .error, message: "(WebSockets transport) WebSocket is nil")
+                return 
+            }
+            
+            do {
+                while !Task.isCancelled {
+                    let message = try await websocket.receive()
+
+                    switch message {
+                        case .string(let text):
+                            logger.log(level: .debug, message: "(WebSockets transport) Received message: \(text)")
+                            await onReceive?(.string(text))
+                        case .data(let data):
+                            await onReceive?(.data(data))
+                    }
                 }
+            } catch {
+                logger.log(level: .error, message: "Failed to receive message: \(error)")
```

Comment:
Maybe we should swallow this error and leave it to URLSesseion Error Callback. In most cases, it's not an error